### DISCUSSION TITLE
Function from 1.4.3 refuses to work in 2.0.1

### DISCUSSION THREAD

**Discussion by ScuttleSE** [savonet/liquidsoap#2111] [User Support]
I have been using the example code from https://www.liquidsoap.info/doc-2.0.0/harbor.html to write a simple authenticator function for my livestream. After upgrading from 1.4.3 to 2.0.1 it has stopped working, and I can't for the life of me figure out why...

Here's the example from the previously mentioned URL

```
def auth(user,password) = 
  # Call an external process to check 
  # the credentials:
  # The script will return the string 
  # "true" of "false"
  #
  # First call the script. Make sure to apply proper escaping
  # of the arguments to prevent command injection!
  ret = process.read.lines("/path/to/script \
         --user=#{user} --password=#{password}")
  # Then get the first line of its output
  ret = list.hd(default="",ret)
  # Finally returns the boolean represented 
  # by the output (bool_of_string can also 
  # be used)
  if ret == "true" then
    true
  else
    false
  end
end
```

And here is my version of it

```
def auth_gbsfm(user, password) = 
  ret = process.read.lines("python3 /home/liq/streampw.py #{password}")
  ret = list.hd(default="",ret)
  if ret == "true" then
    true
  else
    false
  end
end
```

It is more or less exactly the same thing

The error I get is

```
At /home/liq/ices.liq, line 27, char 48-58:
Error 5: this value has type
  (_, _) -> _ (inferred at /home/liq/ices.liq, line 14 char 0 - line 22 char 3)
but it should be a subtype of the type of the value at /usr/share/liquidsoap/2.0.1/libs/telnet.liq, line 73, char 26-47
  ((_) -> bool)?
```

I have a bunch of other stuff in my scriptfile too, but this is the only one that gives me an error. If I comment out the auth function the lic-file runs just fine.

**Comment by toots** [comment]
Hi! The type for the harbor authentication function has changed. We have just added an explanation here: https://www.liquidsoap.info/doc-2.0.0/migrating.html

**Reply by ScuttleSE** [reply]
Updating the example at https://www.liquidsoap.info/doc-2.0.0/harbor.html would be prudent too I guess then....

**Reply by ScuttleSE** [reply]
https://www.liquidsoap.info/doc-2.0.1/harbor.html - and there too I guess

**Reply by toots** [reply]
Ho yeah thanks for that will do!

**Reply by ScuttleSE** [reply]
Kinda hard to debug if the official documentation points you in the wrong direction :)

**Reply by toots** [reply]
Agreed. Doc is fixed and building rn. If you ever have a similar issue feel free to report again and refer to the language reference: https://www.liquidsoap.info/doc-2.0.1/reference.html this one is auto-generated and always up-to date.
